### PR TITLE
Supersimple/allow hex api config

### LINKED ISSUE #660: Setting `api_key` fails with "Invalid key"
With hex 0.19.0 it's impossible to set `api_key` by calling `mix hex.config`, although this option is documented:
```
$ mix hex.config api_key "123"
** (Mix) Invalid key api_key
```

It seems this was introduced by https://github.com/hexpm/hex/commit/9f1cfb05483afa5f4b2e947834354acb21980a20#diff-4394a731980d50a5ccbba9dcdda7bdbdL49
Is this behaviour intended?

### PR DESCRIPTION
**Pull request by supersimple**
🤞Fixes #660 

@ericmj @wojtekmach Please double/triple check this to make sure I got it right.

### CODE REVIEW

**Comment by ericmj** [comment]
It doesn't look correct. `api_key_write_unencrypted` should never be exposed to the user, the name the user should see for this key is `api_key`.

`api_key_read` and `api_key_write` are internal should the user should never see them, the key they are stored with in the config start with `$`, which means they are internal.

**Comment by ericmj** [comment]
To further clarify. The value under `:config` in this map [1] should be the user visible name for the configuration key. If the value in `:config` starts with `$` it should not be visible. If there is no `:config` the config value it is either read-only or can only be set via environment variables.

In theory the map in `Hex.State` should be enough so that we don't need the duplication in `@valid_write_keys` and `@valid_read_keys`.

[1] https://github.com/hexpm/hex/blob/master/lib/hex/state.ex#L9

**Comment by supersimple** [comment]
@ericmj @wojtekmach I think this PR is ready to re-review.
Wojtek and I found another similar issue in the State module where it does not support `api_key`. Specifically, `load_config_value/3`

I think @wojtekmach was going to put up a PR for that.

File: lib/mix/tasks/hex.config.ex [hunk 1]
```diff
@@ -206,10 +190,38 @@ defmodule Mix.Tasks.Hex.Config do
...
   end
 
-  defp label(key) do
-    case key do
-      "api_key_write_unencrypted" -> "api_key"
-      _ -> key
-    end
+  defp valid_keys() do
+    Hex.State.config()
+    |> Enum.map(fn {key, v} ->
```
Comment (ericmj) [inline lib/mix/tasks/hex.config.ex]: I generally try to avoid single pipes, specially if it fits in one line anyway. No deal breaker.

**Comment by wojtekmach** [comment]
> Wojtek and I found another similar issue in the State module where it does not support `api_key`. Specifically, `load_config_value/3` I think @wojtekmach was going to put up a PR for that.

Yes, that's the missing piece - currently we have:

```
$ mix hex.config api_key foo # works, saves to ~/.hex/hex.config
$ mix hex.config api_key
nil
```

I'll look into it later, but if you fix it in the meantime - by all means :)

**Comment by supersimple** [comment]
@wojtekmach ok, i am not sure when i'll have a block of time, but i'll ping you if i start working on it.
I think I will hold off on merging this PR until we get that fixed.

**Comment by wojtekmach** [comment]
Sounds good, same here.

**Comment by supersimple** [comment]
@wojtekmach @ericmj I pushed a change to load_config_value/3 that should resolve this issue. Can you please check it out when you get a chance?

**Comment by ericmj** [comment]
Seems I broke the builds, I will fix it later when I have my computer.

**Comment by supersimple** [comment]
@wojtekmach thanks for the test, it is working now.

File: lib/mix/tasks/hex.config.ex [hunk 2]
```diff
@@ -185,7 +185,8 @@ defmodule Mix.Tasks.Hex.Config do
   defp print_value(_key, value, false, _source), do: Hex.Shell.info(inspect(value, pretty: true))
 
   defp delete(key) do
-    Hex.Config.remove([String.to_atom(key)])
+    {:ok, config_key} = Keyword.fetch(valid_write_keys(), String.to_existing_atom(key))
+    Hex.Config.remove([String.to_existing_atom(config_key)])
```
Comment (ericmj) [inline lib/mix/tasks/hex.config.ex]: You can use `String.to_atom/1` in Hex, doesn't matter for CLI applications imo.

Comment (supersimple) [reply lib/mix/tasks/hex.config.ex]: Ha. I suppose you're right. they'll only be filling up their own atoms table. Force of habit.

**Pull request merged**